plugins/inputs/supervisor/supervisor.go

```diff
@@ -0,0 +1,170 @@
+  ## than you have to add credentials to url (ex. http://login:pass@localhost:9001/RPC2)
+  # url="http://localhost:9001/RPC2"
+  ## Use supervisor identification string as server tag
+  use_identification_tag = false
+  ## Gather PID of running processes
+  gather_pid = false
+  ## Gather exit codes of processes
+  gather_exit_code = false
+`
+
+func (s *Supervisor) Description() string {
+	return "Gather info about processes state, that running under supervisor using its XML-RPC API"
+}
+
+func (s *Supervisor) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Supervisor) Gather(acc telegraf.Accumulator) error {
+	//Initializing XML-RPC client
+	client, err := xmlrpc.NewClient(s.Server, nil)
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	//API call to get information about all running processes
+	var rawProcessData []ProcessInfo
+	err = client.Call("supervisor.getAllProcessInfo", nil, &rawProcessData)
+	if err != nil {
+		return err
+	}
+
+	//API call to get information about instance status
+	err = client.Call("supervisor.getState", nil, &s.Status)
+	if err != nil {
+		return err
+	}
+
+	//API call to get identification string
+	err = client.Call("supervisor.getIdentification", nil, &s.Status.Ident)
+	if err != nil {
+		return err
+	}
+
+	//Iterating through array of structs with processes info and adding fields to accumulator
+	for _, process := range rawProcessData {
+		processTags, processFields, err := s.parseProcessData(process)
+		if err != nil {
+			return err
+		}
+		acc.AddFields("supervisor_processes", processFields, processTags)
+	}
+	// Adding instance info fields to accumulator
+	instanceTags, instanceFields, err := s.parseInstanceData()
+	if err != nil {
+		return err
+	}
+	acc.AddFields("supervisor_instance", instanceFields, instanceTags)
+	return nil
+}
+
+func (s *Supervisor) parseProcessData(pInfo ProcessInfo) (map[string]string, map[string]interface{}, error) {
+	var err error
+	tags := map[string]string{
+		"process": pInfo.Name,
+		"group":   pInfo.Group,
+	}
+	fields := map[string]interface{}{
+		"uptime": pInfo.Now - pInfo.Start,
+		"state":  pInfo.State,
+	}
+	if s.PidGather {
+		fields["pid"] = pInfo.Pid
+	}
+	if s.ExitCodeGather {
+		fields["exitCode"] = pInfo.ExitStatus
+	}
+	if s.UseIdentTag {
+		tags["server"] = s.Status.Ident
+	} else {
+		tags["server"], err = beautifyServerString(s.Server)
+		if err != nil {
+			return map[string]string{}, map[string]interface{}{}, err
+		}
+	}
+	return tags, fields, nil
+}
+
+func (s *Supervisor) parseInstanceData() (map[string]string, map[string]interface{}, error) {
+	var err error
+	tags := make(map[string]string, 1)
+	fields := make(map[string]interface{}, 1)
+	if s.UseIdentTag {
+		tags["server"] = s.Status.Ident
+	} else {
+		tags["server"], err = beautifyServerString(s.Server)
+		if err != nil {
+			return map[string]string{}, map[string]interface{}{}, err
+		}
+	}
+	fields["state"] = s.Status.StateCode
+	return tags, fields, nil
```

Comment:
Maybe this is a bit shorter
```suggestion
	server := s.Status.Ident
	if !s.UseIdentTag {
		var err error
		server, err = beautifyServerString(s.Server)
		if err != nil {
			return nil, nil, err
		}
	}
	tags := map[string]string{"server": server}
	fields := map[string]interface{}{"state": s.Status.StateCode}
	return tags, fields, nil
```

Reply:
Will be included in next commit